codec-http/src/main/java/io/netty/handler/codec/http/websocketx/WebSocketServerHandshakerFactory.java

```diff
@@ -148,18 +148,23 @@ public static void sendUnsupportedWebSocketVersionResponse(Channel channel) {
      * Return that we need cannot not support the web socket version
      */
     public static ChannelFuture sendUnsupportedVersionResponse(Channel channel) {
-        return sendUnsupportedVersionResponse(channel, channel.newPromise());
+        HttpResponse res = new DefaultFullHttpResponse(
+                HttpVersion.HTTP_1_1,
+                HttpResponseStatus.UPGRADE_REQUIRED);
+        res.headers().set(HttpHeaderNames.SEC_WEBSOCKET_VERSION, WebSocketVersion.V13.toHttpHeaderValue());
+        HttpUtil.setContentLength(res, 0);
```

Comment:
Dup code. Couldn’t this response be a constant ?

Reply:
yep it could...